tasks/parse-pdr/index.js

```diff
@@ -36,6 +37,17 @@ function parsePdr(event) {
   return parse.ingest()
     .then((payload) => {
       if (parse.connected) parse.end();
+      // opportunity to filter the granules of interest based on regex in granuleIdExtraction
+      if (config && config.collection && config.collection.granuleIdExtraction) {
+        const granuleIdExtraction = config.collection.granuleIdExtraction;
+        const granules = payload.granules.filter((g) => g.files[0].name.match(granuleIdExtraction));
+        const filteredPayload = {
+          granules: granules,
+          granulesCount: granules.length
+        };
+        return Object.assign({}, event.input, filteredPayload);
```

Comment:
This is performing a shallow clone on `event.input`.  A more robust option would be to use Lodash's `cloneDeep` function.

`return Object.assign(cloneDeep(event.input, filteredPayload);`

Reply:
ok